artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/ConcurrentLongHashMap.java

```diff
@@ -457,11 +459,11 @@ private void rehash() {
             }
          }
 
-         capacity = newCapacity;
          keys = newKeys;
          values = newValues;
          usedBuckets = size;
-         resizeThreshold = (int) (capacity * MapFillFactor);
+         capacityUpdater.lazySet(this, newCapacity);
+         resizeThreshold = (int) (newCapacity * MapFillFactor);
```

Comment:
`capacityUpdater.lazySet` is store-releasing `keys` , `values`, `usedBuckets`, but not `resizeThreshold`: that means that the `resizeThreshold` store could be happen before the update of the `capacity`, is it correct?